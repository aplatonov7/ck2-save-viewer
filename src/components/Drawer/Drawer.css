.Drawer__delete-btn {
  border: none;
  background: none;
  cursor: pointer;
  padding: .2rem .5rem;
  background-color: #eee;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);
  font-size: 0;
}

.Drawer__delete-btn:after {
  content: 'x';
  font-size: 1rem;
}

.Drawer__row {
  display: flex;
  justify-content: space-between;
}

.Drawer {
  position: absolute;
  top: 0;
  left: 0;
  bottom: 0;
  width: 500px;
  background-color: #fff;
  box-shadow: 0 1px 3px rgba(0,0,0,0.12), 0 1px 2px rgba(0,0,0,0.24);

  padding: 1rem;

  transform: translateX(-502px);
  transition: transform .2s ease; 
}

.Drawer.Drawer__opened {
  transform: translateX(0);
}