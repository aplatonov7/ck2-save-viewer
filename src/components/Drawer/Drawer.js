import React from 'react'
import PropTypes from 'prop-types'
import cn from 'classnames'

import './Drawer.css'

const Drawer = ({ opened, onClear }) => (
  <aside className={cn('Drawer', { Drawer__opened: opened })}>
    <div className="Drawer__row">
      Data is loaded
      <button
        className="Drawer__delete-btn"
        title="Delete current data so you can upload another file"
        onClick={onClear}
      >
        Delete data
      </button>
    </div>
  </aside>
)

Drawer.propTypes = {
  opened: PropTypes.bool.isRequired,
  onClear: PropTypes.func.isRequired,
}

export default Drawer
