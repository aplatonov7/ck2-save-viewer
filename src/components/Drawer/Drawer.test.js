import React from 'react'
import Drawer from './Drawer'

describe('Drawer component', () => {
  let wrapper

  const props = {
    opened: false,
    onClear: jest.fn(),
  }

  beforeEach(() => {
    wrapper = shallow(<Drawer {...props} />)
  })

  it('should render a button that triggers onClear prop', () => {
    const btn = wrapper.find('.Drawer__delete-btn')
    expect(btn.length).toBe(1)
    btn.simulate('click')
    expect(props.onClear.mock.calls.length).toBe(1)
  })

  it('should add Drawer__opened class to root element when opened prop is true', () => {
    expect(wrapper.hasClass('Drawer__opened')).toBe(false)
    wrapper.setProps({ opened: true })
    expect(wrapper.hasClass('Drawer__opened')).toBe(true)
  })

  it('should match snapshot', () => {
    const tree = renderer.create(<Drawer {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
