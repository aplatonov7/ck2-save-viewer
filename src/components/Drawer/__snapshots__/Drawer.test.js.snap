// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Drawer component should match snapshot 1`] = `
<aside
  className="Drawer"
>
  <div
    className="Drawer__row"
  >
    Data is loaded
    <button
      className="Drawer__delete-btn"
      onClick={[Function]}
      title="Delete current data so you can upload another file"
    >
      Delete data
    </button>
  </div>
</aside>
`;
